src/manifest/classes/period.ts

```diff
@@ -135,6 +98,30 @@ export default class Period implements IPeriodMetadata {
     this.streamEvents = args.streamEvents === undefined ? [] : args.streamEvents;
   }
 
+  createAdaptationsObject(
```

Comment:
Is the intention to create a private method that is only called at the start of the Period's instantiation?

If it is, I have some issues:
  -  it's not marked as a private method
  - I tend to think that it is a bad idea to call a method before the instantiation was succesfully done because we're in a partial state in which even TypeScript might not see that the current instance is not fully constructed / initialized. It's a difficult state to think about.

As we're not even relying on the Period's context it seems here, can't we just make that a regular function?